little_raft/src/timer.rs

```diff
@@ -9,30 +9,27 @@ pub struct Timer {
 // Timer fires after the specified duration. The timer can be renewed.
 impl Timer {
     pub fn new(timeout: Duration) -> Timer {
-        let (tx, rx) = bounded(1);
-        thread::spawn(move || {
-            thread::sleep(timeout);
-            let _ = tx.send(());
-        });
-
         Timer {
             timeout: timeout,
-            rx: rx,
+            rx: get_timeout_channel(timeout),
         }
     }
 
     pub fn renew(&mut self) {
-        let (tx, rx) = bounded(1);
-        let timeout = self.timeout;
-        thread::spawn(move || {
-            thread::sleep(timeout);
-            let _ = tx.send(());
-        });
-
-        self.rx = rx;
+        self.rx = get_timeout_channel(self.timeout);
     }
 
     pub fn get_rx(&self) -> &Receiver<()> {
         &self.rx
     }
 }
+
+fn get_timeout_channel(timeout: Duration) -> Receiver<()> {
```

Comment:
Let's move this function into `impl Timer` and just keep it private and call it with `Timer::get_timeout_channel(self.timeout)`.